Added Medical Patches

jorgun opened the pull request:
<!-- Please read these guidelines before opening your PR: https://docs.spacestation14.io/en/getting-started/pr-guideline -->
<!-- The text between the arrows are comments - they will not be visible on your PR. -->

## About the PR
<!-- What did you change in this PR? -->
This PR adds Medical Patches a way to add chems over time, and in larger quanta then a syringe,
there are 4 flavors. basic rapid large, and makeshift patch.
fill the patches in the chemmaster and stick them to teh patient to transfer the chemicals.
when the patch is empty it will fall off on its own.

the makeshift is a bit different, it can be crafted by hand, can be filled by hand and is becomes a trash item after use

[ref: patch ](https://www.paradisestation.org/wiki/index.php?title=Medical_Items)

**TODO**:
- [x] add Sprites
- [x] add some prefilled patches to nanomed and firstaid kit
- [x] add a bounty
- [x] add stupid localisation

## Why / Balance
<!-- Why was it changed? Link any discussions or issues here. Please discuss how this would affect game balance. -->
There have been a few PR over on wizden where they have been trying to figure out how to make people not abuse the Chemmaster,  by adding this there is anoter reason to not fill the chemmaster, or atlest keep one of them empty.
and gives the cemists/doctors a new tool.

this is atlso a bit more advanced tool for the chemists as mixing the content whitout overdoisng requires **math**

## Technical details
<!-- If this is a code change, summarize at high level how your new code works. This makes it easier to review. -->
changed stickysystem as you cant remove the patch from self whitout trigering the unstick prossess.

the basic holds 40u of chems, 1u of chems transfered evry 1 sec
the rapid holds 40u of chems, 2u of chems transfered evry 1 sec
the large holds 60uof chems,1u of chems transfered evry 1 sec and 15% of chems are transfered on use (10u if full)
makesihft holds 15u and transfer 1u evry 2 sec.
Prefilled from nanomed, holds 30u or reagents injects 0.25u a sec, cant be emptied, cant be filled, and is destroyed when removed/empty  
## Media
<!-- 
PRs which make ingame changes (adding clothing, items, new features, etc) are required to have media attached that showcase the changes.
Small fixes/refactors are exempt.
Any media may be used in SS14 progress reports, with clear credit given.

If you're unsure whether your PR will require media, ask a maintainer.
If you're unsure whether your PR will require media, ask a maintainer.

Check the box below to confirm that you have in fact seen this (put an X in the brackets, like [X]):
-->

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

![image](https://github.com/user-attachments/assets/76dd97fd-e62d-4c90-9c8c-9f02953e8b21)
patchse, basic rapid and large,
prefilled, derma bicardine and used
makeshift  and used makeshift
![image](https://github.com/user-attachments/assets/eb2f312b-18bd-44dc-b7cb-1eff62e39704)
bit large innit?
![image](https://github.com/user-attachments/assets/95af4e9f-ee30-4296-9964-3fbd7f389862)


## Breaking changes
<!--
List any breaking changes, including namespace, public class/method/field changes, prototype renames; and provide instructions for fixing them. This will be pasted in #codebase-changes.
-->

**Changelog**
<!--
Make players aware of new features and changes that could affect how they play the game by adding a Changelog entry. Please read the Changelog guidelines located at: https://docs.spacestation14.io/en/getting-started/pr-guideline#changelog
-->

<!--
Make sure to take this Changelog template out of the comment block in order for it to show up. Changelog must have a :cl: symbol, so the bot recognizes the changes and adds them to the game's changelog.

- remove: Removed fun!
- tweak: Changed fun!
- fix: Fixed fun!
-->
:cl: (jorgun , Fishbait_x , Huffs-the-Frezon)
- add: Added Medical Patches

whateverusername0 commented:
holy based

whateverusername0 commented on Content.Server/Goobstation/MedicalPatch/MedicalPatchComponent.cs:
```diff
@@ -0,0 +1,25 @@
+namespace Content.Server.Medical.Components;
+
+[RegisterComponent]
+[AutoGenerateComponentState]
```
i believe you don't need it if you have default values everywhere

whateverusername0 commented on Content.Server/Goobstation/MedicalPatch/MedicalPatchComponent.cs:
```diff
@@ -0,0 +1,25 @@
+    [DataField] // [ViewVariables(VVAccess.ReadWrite)]
+    public FixedPoint2 TransferAmount = FixedPoint2.New(1);
+    [DataField]
+    public bool SingelUse = false;
```
minor spelling mistake

whateverusername0 commented on Content.Server/Goobstation/MedicalPatch/MedicalPatchSystem.cs:
```diff
@@ -0,0 +1,124 @@
+    {
+        if (component.SingelUse)
+        {
+            //_polymorph.PolymorphEntity(uid, "UsedMedicalPatch");
```
if reagents are empty you can remove the existing entity and add a new one instead of polymorphing into it.
should be much easier and better

whateverusername0 replied:
@jorgun you can spawn it simply by doing `Spawn("UsedMedicalPatch", Transform(uid).Coordinated);`
pls do sth 🥺

jorgun replied:
diddnt quite do the trick but by copinyg how snacks gets turned into trash i got it working

whateverusername0 commented on Content.Server/Sticky/Systems/StickySystem.cs:
```diff
@@ -187,7 +188,7 @@ public void StickToEntity(EntityUid uid, EntityUid target, EntityUid user, Stick
             _appearance.SetData(uid, StickyVisuals.IsStuck, true, appearance);
         }
-
+        EnsureComp<UnremoveableComponent>(uid); //Goobstation - MedicalPatch
```
maybe don't? just make it when it gets unattached to stop injecting reagents or to get turned into a wasted one immediately
this shit WILL fuck up plungers. i believe.

jorgun replied:
Plunger dont use the stickysystem, only C4 uses the stickysystem, 
there is a small bug/feature that when you place a c4 on yourself you can just pick it up from the new container created insted of unsticking it.

whateverusername0 replied:
> Plunger dont use the stickysystem, only C4 uses the stickysystem, there is a small bug/feature that when you place a c4 on yourself you can just pick it up from the new container created insted of unsticking it.

oh ok, doesn't matter then, i'm sure nothing will break whatsoever :trollface:

whateverusername0 commented on Resources/Prototypes/Goobstation/Entities/Objects/Specific/Medical/medical_patch.yml:
```diff
@@ -0,0 +1,156 @@
+      - MobState  # only atatch to living
+    blacklist:
+      components:
+      - BorgChassis # borgs are alive but dont get band aids.
```
this can be replaced by a Body component i believe. will make every possible living object except borgs be patchable

jorgun replied:
giving medical patches to the pets i think shud be allowed, anything else like borgs guardians or anything else the patch will fall off due to no bloodstream

whateverusername0 replied:
speaking of bloodstream, you can check for that component instead of using whatever you have now :godo:

whateverusername0 commented on Resources/Prototypes/Goobstation/Entities/Objects/Specific/Medical/medical_patch.yml:
```diff
@@ -0,0 +1,156 @@
+  parent: BaseHealingItem
+  id: UsedMedicalPatch
+  name: A Used Meical Patch
+  description: EEEEWWWWWWW
```
how silly and awesome EEEEEEWWWWWWWWW 😨

Aidenkrz replied:
meical patch

jorgun replied:
yes. spelling

whateverusername0 commented on Resources/Prototypes/Goobstation/Entities/Objects/Specific/Medical/medical_patch.yml:
```diff
@@ -0,0 +1,156 @@
+  parent: MedicalPatchBasic
+  id: MedicalPatchMakeshift
+  name: "Makeshift Patch"
+  description: does not look hygenic
```
minor spelling mistake

whateverusername0 commented on Resources/Prototypes/Goobstation/Entities/Objects/Specific/Medical/medical_patch.yml:
```diff
@@ -0,0 +1,156 @@
+- type: entity
+  parent: MedicalPatchBasic
+  id: MedicalPatchMakeshift
+  name: "Makeshift Patch"
```
you don't have to put them in braces

whateverusername0 commented on Resources/Prototypes/Goobstation/Entities/Objects/Specific/Medical/medical_patch.yml:
```diff
@@ -0,0 +1,156 @@
+- type: entity
+  parent: BaseHealingItem
+  id: MedicalPatchBasic
+  name: "Medical patch"
```
you don't have to put them in braces
also no big letters in names

whateverusername0 reviewed (changes requested):
made a quick review please check

whateverusername0 reviewed (approved):
ok looks good enough to me :+1:

jorgun commented:
> ok looks good enough to me 👍

i think so to, now we just need to do anoter round of balancing

whateverusername0 commented:
> i think so to, now we just need to do anoter round of balancing

also i suggest you find sprites more resembling a medical patch not just a bangade :trollface:

jorgun commented:
> > i think so to, now we just need to do anoter round of balancing
> 
> also i suggest you find sprites more resembling a medical patch not just a bangade :trollface:

TODO!

Piras314 reviewed (changes requested):
Make a BaseMedicalPatch that MedicalPatchBasic and MedicalPatchPrefilledBase are children of, and change the popup messages to "Started placing patch" "Patch placed" "Started removing patch" "Patch removed" (which conveniently you will only have to do once in BaseMedicalPatch now!

jorgun commented:
changed base, 
added new sprite , added a solution indicator bar that colors after the soulution
added a border layer that can be used to tell the patches apart.

jorgun commented:
> don't see the push / commit rn

i forgot to push

jorgun commented:
![image](https://github.com/user-attachments/assets/574ccffc-de27-4c0f-b461-ed9286ea37b5)
working on new sprite

whateverusername0 commented:
> working on new sprite

yeah no making it stick onto you like a c4 is a bad idea
what if no sprite at all? or some kind of bandage like in visual traits (that will not be seen through hardsuits ofc)

JoeHammad1844 commented:
bandaid sprite is super blurry

jorgun commented:
> > working on new sprite
> 
> yeah no making it stick onto you like a c4 is a bad idea what if no sprite at all? or some kind of bandage like in visual traits (that will not be seen through hardsuits ofc)

Can make em not "visible" by not using sticky visuliser component. They will be behind the character On dreawdepth 3. There is a bug when you use a patch on your self, it don't trigger sticky visualiser. But as that is a self attached I don't mind the patch being hidden. 
I think they shud be visible to indicate a transfer of chemicals.  
I feel the sticky system works excellent the way it is now.

jorgun commented:
> bandaid sprite is super blurry

Bandaid is currently only used for Makeshift and as Trash item. Il respite it when I get a better idea for how a Makeshift shud look like.
